### PR TITLE
Added error handling

### PR DESCRIPTION
**Pull request by danishjamal104**
## Description

* Intergrated interop error code handling
* Minor logical bug resolved

 _(Ignore if these details are present on the associated JIRA ticket)_

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Design related bullet points or design document link related to this PR added in the description above. 

- [x] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [x] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

File: src/main/java/org/mifos/connector/ams/errorhandler/ErrorTranslator.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        String errorCode = (String) zeebeFinalVariables.get(ERROR_CODE);
+        if (errorCode == null) {
+            return zeebeFinalVariables;
+        }
+
+        PaymentHubError paymentHubError;
+        try {
+            paymentHubError = errorMapper.getInternalError(errorCode);
+            if (paymentHubError == null) {
+                throw new NullPointerException();
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/ams/errorhandler/ErrorTranslator.java:34]: create a generic error for this.

File: src/main/java/org/mifos/connector/ams/errorhandler/ErrorTranslator.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                    .build();
+        }
+
+
+        zeebeFinalVariables.put(ERROR_CODE, phErrorDTO.getErrorCode());
+        zeebeFinalVariables.put(ERROR_INFORMATION, phErrorDTO.getErrorDescription());
+        try {
+            zeebeFinalVariables.put(ERROR_INFORMATION, objectMapper.writeValueAsString(phErrorDTO));
+            PhErrorDTO errorDTO = objectMapper.readValue((String) zeebeFinalVariables.get(ERROR_INFORMATION), PhErrorDTO.class);
+            System.out.println("I got no error");
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/ams/errorhandler/ErrorTranslator.java:64]: remove system.out.println. use loggers

File: src/main/java/org/mifos/connector/ams/interop/ClientResponseProcessor.java [hunk 3]
```diff
@@ -90,7 +80,7 @@ public void process(Exchange exchange) throws Exception {
             if ("1.2".equals(amsVersion)) {
                 ClientData client = exchange.getIn().getBody(ClientData.class);
                 EnumOptionData legalForm = client.getLegalForm();
-                if (legalForm == null || PERSON.equals(LegalForm.valueOf(legalForm.getValue()))) {
+                if (legalForm == null || PERSON.equals(LegalForm.valueOf(legalForm.getValue().toUpperCase()))) {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/ams/interop/ClientResponseProcessor.java:83]: check null pointers

File: src/main/java/org/mifos/connector/ams/interop/ErrorParserRouteBuilder.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public void configure() {
+
+        // parses error payload from fineract and sets relevant error related variable if error exist
+        from("direct:error-handler")
+                .id("error-handler")
+                .log(LoggingLevel.INFO, "Error handler for response ${body}")
+                .choice()
+                // check if http status code is <= 202
+                .when(e -> e.getIn().getHeader(Exchange.HTTP_RESPONSE_CODE, Integer.class) >= 202)
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/ams/interop/ErrorParserRouteBuilder.java:36]: we have to check the other 2** series success responses as well and ignore them here.

File: src/main/java/org/mifos/connector/ams/interop/InteroperationRouteBuilder.java [hunk 5]
```diff
@@ -101,10 +107,36 @@ public void configure() {
...
+                    } catch (TenantNotExistException e) {
+                        e.printStackTrace();
+                        exchange.setProperty(ERROR_CODE, PaymentHubError.PayeeFspNotConfigured.getErrorCode());
+                        exchange.setProperty(ERROR_INFORMATION, PaymentHubError.PayeeFspNotConfigured.getErrorDescription());
+                        exchange.setProperty(ERROR_PAYLOAD, PaymentHubError.PayeeFspNotConfigured.getErrorDescription());
+                        exchange.getIn().setHeader(Exchange.HTTP_RESPONSE_CODE, 404);
+                        exchange.setProperty(IS_ERROR_SET_MANUALLY, true);
+                    }
+                })
                 .log("Test 1: ${body}")
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/ams/interop/InteroperationRouteBuilder.java:123]: update this log statement

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/connector/ams/errorhandler/ErrorTranslator.java (hunk 1 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/connector/ams/errorhandler/ErrorTranslator.java:35]: on hold, and added todo

File: src/main/java/org/mifos/connector/ams/errorhandler/ErrorTranslator.java (hunk 2 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/connector/ams/errorhandler/ErrorTranslator.java]: done

File: src/main/java/org/mifos/connector/ams/interop/ClientResponseProcessor.java (hunk 3 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/connector/ams/interop/ClientResponseProcessor.java]: done

File: src/main/java/org/mifos/connector/ams/interop/ErrorParserRouteBuilder.java (hunk 4 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/connector/ams/interop/ErrorParserRouteBuilder.java]: done

File: src/main/java/org/mifos/connector/ams/interop/InteroperationRouteBuilder.java (hunk 5 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/connector/ams/interop/InteroperationRouteBuilder.java]: done

**Pull request merged**